packages/core/src/route/types.ts

```diff
@@ -85,19 +89,22 @@ type IDecoratorMetadata<T> = {
  * @param typeFactory - Function that returns a type (useful for lazily evaluated types)
  * @param required - Whether the property is required or not
  */
-export interface IPropDecoratorOptions extends Omit<IJsonSchema, 'type' | 'required'> {
-	type?: TypeValue;
+export type IPropDecoratorOptions<T> = Omit<JSONSchemaType<T, true>, 'required' | 'type'> & {
+	type?: TypeValue | JSONSchemaType<T>['type'];
```

Comment:
out of curiosity, what's the benefit of passing the generic type to `JSONSchemaType`?

Reply:
Example here: https://github.com/HPInc/davinci/pull/51/files#diff-5db34c11902eb9d0183002af2ec1ab37668bd589c46e606d6d7281952c4cf40cR25
If I specified the generic type, it won't allow me to add non-related options (e.g. `oneOf: [ { type: 'object' } ]` won't be accepted).